libs/libcommon/src/libcommon/queue.py

```diff
@@ -544,11 +530,10 @@ def cancel_jobs_by_job_id(self, job_ids: list[str]) -> int:
         """
         try:
             existing = JobDocument.objects(pk__in=job_ids)
-            previous_status = [(job.type, job.status) for job in existing.all()]
-            existing.update(finished_at=get_datetime(), status=Status.CANCELLED)
-            for job_type, status in previous_status:
-                update_metrics_for_type(job_type=job_type, previous_status=status, new_status=Status.CANCELLED)
-            return existing.count()
+            for job in existing.all():
+                decrease_metric(job_type=job.type, status=job.status)
```

Comment:
maybe not very important, but instead of N -1 operations, we could do -N directly by grouping by job type.

maybe in another pr